src/app/rpmostree-builtin-kargs.c

```diff
@@ -62,6 +60,118 @@ get_kargs_option_variant (void)
+                          const char    **out_kernel_arg,
+                          GCancellable   *cancellable,
+                          GError        **error)
+{
+  g_autofree char *chomped_input = g_strchomp (g_strdup (input_kernel_arg));
+
+  __attribute__((cleanup(_ostree_kernel_args_cleanup))) OstreeKernelArgs *temp_kargs = _ostree_kernel_args_from_string (chomped_input);
+
+  /* We check existance of ostree argument, if it
+   * exists, we directly remove it. Also Note, since the current kernel
+   * arguments are directly collected from the boot config, we expect that
+   * there is only one value associated with ostree argument. Thus deleting
+   * by one value should not error out here.
+   */
+  if (_ostree_kernel_args_get_last_value (temp_kargs, "ostree") != NULL )
+    {
+      if (!_ostree_kernel_args_delete (temp_kargs, "ostree", error))
+        return FALSE;
+    }
+  g_autofree char* filtered_input = _ostree_kernel_args_to_string (temp_kargs);
+
+  g_autofree char *input_string = g_strdup_printf ("\n"
+      "# Please enter the kernel arguments. Each kernel argument"
+      "# should be in the form of key=value.\n"
+      "# Lines starting with '#' will be ignored. Each key=value pair should be \n"
+      "# separated by spaces, and multiple value associated with one key is allowed. \n"
+      "# Also, please note that any changes to the ostree argument will not be \n"
+      "# effective as they are usually regenerated when bootconfig changes. \n"
+      "%s", filtered_input);
+
+  /* Note: the repo here is NULL, as we don't really require it
+   * for the edtior process
+   */
+  g_autofree char *out_editor_string = ot_editor_prompt (NULL, input_string,
+                                                         cancellable, error);
+  if (out_editor_string == NULL)
+    return FALSE;
+
+  /* We now process the editor output, we ignore empty lines and lines
+   * starting with '#'
+   */
+  g_auto(GStrv) lines = g_strsplit (out_editor_string, "\n", -1);
+  guint num_lines = g_strv_length (lines);
+  g_autoptr(GString) kernel_arg_buf = g_string_new ("");
+  for (guint i = 0; i < num_lines; i++)
+    {
+      char *line = lines[i];
+      /* Remove leading and trailing spaces */
+      g_strstrip (line);
+
+      /* Comments and blank lines are skipped */
+      if (*line == '#')
+        continue;
+      if (*line == '\0')
+        continue;
+      g_string_append (kernel_arg_buf, line);
+
+      /* We append a space to every line except the last
+       * because there could be a case where user have
+       * args that are in multiple lines.
+       */
+      g_string_append (kernel_arg_buf,  " ");
+    }
+
+  /* Transfer the ownership to a new string, so we can remove the trailing spaces */
+  g_autofree char *kernel_args_str = g_string_free (g_steal_pointer(&kernel_arg_buf), FALSE);
+  g_strchomp (kernel_args_str);
+
+  g_auto(GStrv) kernel_arg_list = g_strsplit (kernel_args_str, " ", -1);
```

Comment:
What I meant here was to just use `_ostree_kernel_args_from_string` again and then just query using `_ostree_kernel_args_get_last_value` like we do above?

Reply:
ah I see =)